### DISCUSSION TITLE
IRAM usage error while building custom firmware

### DISCUSSION THREAD

**Discussion by d-solis-tovar** [micropython/micropython#13224] [ESP32]
Hi,

I'm starting a project to build my own custom micropython firmware for esp32. 

I first tried to build the `ESP32_GENERIC` board firmware using `ESP-IDF` and `make` commands. However, I always find the same error: 

`micropython.elf section `.iram0.text' will not fit in region `iram0_0_seg'`

The applications static IRAM usage is larger than the available IRAM size.

I read other posts where it is suggested to modify partition table and configuration settings on sdkconfig file to optimize memory usage. However, none of these solutions worked for me.

I can't figure out what is going wrong. Any suggestions for me to try? 

I'm using `ESP-IDF v5.0.4`

Thank you!

**Comment by LeosDing** [comment]
> Hi,
> 
> I'm starting a project to build my own custom micropython firmware for esp32.
> 
> I first tried to build the `ESP32_GENERIC` board firmware using `ESP-IDF` and `make` commands. However, I always find the same error:
> 
> `micropython.elf section `.iram0.text' will not fit in region `iram0_0_seg'`
> 
> The applications static IRAM usage is larger than the available IRAM size.
> 
> I read other posts where it is suggested to modify partition table and configuration settings on sdkconfig file to optimize memory usage. However, none of these solutions worked for me.
> 
> I can't figure out what is going wrong. Any suggestions for me to try?
> 
> I'm using `ESP-IDF v5.0.4`
> 
> Thank you!

I have found same question ...

**Reply by jimmo** [reply]
Do you have any changes and are you using the latest MicroPython code? We build the `ESP32_GENERIC` firmware as part of the continuous integration on GitHub, so it should work.

**Comment by shariltumin** [comment]
This is my build script "mpb.sh":

```bash
#!/bin/bash

export PATH="$HOME/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"

echo "=========================================================="
echo "Setting up for esp32 build"
echo "Using esp-idf-504"
export IDF_PATH="$HOME/disk/esp/esp-idf-504"
export IDF_TARGET="esp32"
source $IDF_PATH/export.sh

make V=1 BOARD=ESP32_GENERIC
```

You may need to change the paths to suit your local installations.

Put this script in the micropython/ports/esp32 and ```chmod u+x mpb.sh```. Then just run it,
```./mpb.sh``` 

Remember to do ```make -C mpy-cross``` in the micropython directory first.

Hope this helps.

**Comment by GCTechnology** [comment]
The same problem on `ESP-IDF v5.2`, a 'build' command under `./ports/esp32`

**Reply by jonnor** [reply]
esp-idf 5.2 is known to be bigger, see https://github.com/orgs/micropython/discussions/14263 
Using 5.0.4 though should work, at least with a default MicroPython board configuration. All the builds currently on the MicroPython download page are made that way.

**Reply by GCTechnology** [reply]
`ESP-IDF v5.0.4` works.
